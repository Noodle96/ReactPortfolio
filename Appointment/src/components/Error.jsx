import React from 'react'

const Error = ({children}) => {
  return (
    <div className=' bg-red-500 uppercase text-white text-center text-xs p-3 font-bold mb-3 rounded-md'>
        {children}
    </div>
  )
}

export default Error
