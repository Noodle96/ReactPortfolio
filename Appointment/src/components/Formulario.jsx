import React from 'react'

const Formulario = () => {
  return (
    <div className=' md:w-1/2 lg:w-2/5'>
         <h2 className=' text-center font-black text-2xl'>
            Seguimiento Pacientes
        </h2>
        <p className=' text-base mt-5 text-center mb-10'>
            AÃ±ade Pacientes y {''}
            <span className='text-indigo-600 font-black'>Administralos</span>
        </p>
        <form action="#" className='bg-white shadow-md rounded-lg py-10 px-5 mb-10'>
            <div className='mb-5'>
                <label htmlFor='mascota' className='block text-gray-700 uppercase font-bold'>
                    Mascota:
                </label>
                <input
                    id='mascota'
                    type="text"
                    placeholder='Nombre Mascota'
                    className='border-2 w-full p-2 m-2 placeholder-gray-400 rounded-md'
                />
            </div>{/* END ROW */}
            <div className='mb-5'>
                <label htmlFor='propietario' className='block text-gray-700 uppercase font-bold'>
                    Propietario:
                </label>
                <input
                    id='propietario'
                    type="text"
                    placeholder='Nombre del propietario'
                    className='border-2 w-full p-2 m-2 placeholder-gray-400 rounded-md'
                />
            </div>{/* END ROW */}
            <div className='mb-5'>
                <label htmlFor='email' className='block text-gray-700 uppercase font-bold'>
                    Email:
                </label>
                <input
                    id='email'
                    type="email"
                    placeholder='Email del propietario - # Emergencia'
                    className='border-2 w-full p-2 m-2 placeholder-gray-400 rounded-md'
                />
            </div>{/* END ROW */}
            <div className='mb-5'>
                <label htmlFor='alta' className='block text-gray-700 uppercase font-bold'>
                    Fecha Alta:
                </label>
                <input
                    id='alta'
                    type="date"
                    className='border-2 w-full p-2 m-2 placeholder-gray-400 rounded-md'
                />
            </div>{/* END ROW */}
            <div className='mb-5'>
                <label htmlFor='sintomas' className='block text-gray-700 uppercase font-bold'>
                    Sintomas:
                </label>
                <textarea
                    name="" 
                    id="sintomas"
                    rows= '8'
                    className='border-2 w-full p-2 m-2 placeholder-gray-400 rounded-md'
                    placeholder='Describe los sintomas'
                >
                </textarea>
            </div>{/* END ROW */}
            <input
                type="submit"
                value="Agregar Paciente"
                className=' bg-indigo-600 w-full p-3 text-white uppercase
                            hover:bg-indigo-700 font-bold cursor-pointer transition-all'
            />
        </form>
    </div>
  )
}

export default Formulario;