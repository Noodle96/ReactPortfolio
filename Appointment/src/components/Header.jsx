function Header(){
    return(
        <header>
            <h1 >Desde Header</h1>
        </header>
    );
}

export default Header;