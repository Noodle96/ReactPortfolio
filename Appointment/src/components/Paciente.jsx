import React from 'react'

const Paciente = ({paciente,setPacienteToEdit}) => {
    const {mascota, propietario,email, fecha, sintomas} = paciente;
    return (
        <div className='m-3 bg-white shadow-md px-5 py-10 rounded-xl'>
            <p className='font-bold mb-3 text-gray-700 uppercase'>
                Nombre: {' '}
                <span className='font-normal normal-case'>{mascota}</span>
            </p>
            <p className='font-bold mb-3 text-gray-700 uppercase'>
                Propietario: {' '}
                <span className='font-normal normal-case'>{propietario}</span>
            </p>
            <p className='font-bold mb-3 text-gray-700 uppercase'>
                Email: {' '}
                <span className='font-normal normal-case'>{email}</span>
            </p>
            <p className='font-bold mb-3 text-gray-700 uppercase'>
                Fecha Alta: {' '}
                <span className='font-normal normal-case'>{fecha}</span>
            </p>
            <p className='font-bold mb-3 text-gray-700 uppercase'>
                Sintomas Alta: {' '}
                <span className='font-normal normal-case'>
                    {sintomas}
                </span>
            </p>
            <div className='flex justify-between'>
                <button
                    type="button"
                    className=' bg-indigo-600 hover:bg-indigo-700 py-2 px-10 uppercase text-white font-bold rounded-lg'
                    onClick={() => setPacienteToEdit(paciente)}
                >
                    Editar
                </button>
                <button
                    type="button"
                    className=' bg-red-500 hover:bg-red-600 py-2 px-10 uppercase text-white font-bold rounded-lg'
                >
                    Eliminar
                </button>
            </div>
        </div>
    )
}

export default Paciente
