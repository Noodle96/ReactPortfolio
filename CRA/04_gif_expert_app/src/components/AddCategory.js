//rafc
import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value)
        console.log('handleInputChange llamado');
    }
    
    const handleSubmit = (e) =>  {
        e.preventDefault();
        console.log('handleSubmit llamado, ',inputValue);
        // console.log('hecho submit');
        if(inputValue.trim().length > 1){
            setCategories((cats) => [...cats, inputValue]);
            // setCategories([inputValue])
            setInputValue('');
        }
    }
    return (
        <form onSubmit={ (e) => handleSubmit(e)}>
            {/* <h1>{inputValue}</h1> */}
            <p>{inputValue}</p>
            <input 
                type='text'
                value={inputValue}
                onChange={(e) => handleInputChange(e)}
            />
        </form>
    );
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}