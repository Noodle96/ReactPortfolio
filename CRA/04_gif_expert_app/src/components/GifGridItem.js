import React from 'react'
import PropTypes from 'prop-types';
// {id,title,url,type,username}
export const GifGridItem = ({title,url}) => {
    // console.log({id,title,url});
    return (
        <div className='card animate__animated animate__bounce'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>

        // <Fragment>
        //     <div className="wrapper">
        //         <div className="product-img">
        //         <img src={url} alt={title} />
        //         </div>
        //         <div className="product-info">
        //             <div className="product-text">
        //                 <h1 >{title}</h1>
        //                 <h2>{type}</h2>
        //                 <p>{username}  </p>

        //             </div>
        //             <div className="product-price-btn">
        //                  <p><span>Free</span></p>
        //                  <button type="button">Download</button>
        //              </div>
        //         </div>
        //     </div>
        // </Fragment>
    )
}


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}


