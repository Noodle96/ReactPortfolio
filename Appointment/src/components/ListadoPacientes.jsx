import React from 'react'
import Paciente from './Paciente'

const ListadoPacientes = ({pacientes, setPacienteToEdit, eliminarPaciente}) => {

    
    return (
        <div className=' md:w-1/2 lg:w-3/5 md:h-screen md:overflow-y-scroll'>
            {pacientes && pacientes.length ? (
                <>
                    <h2 className='font-black text-3xl text-center'>Listado de Pacientes</h2>
                    <p className='text-xl mt-5 mb-10 text-center'>
                        Adminitra tus{' '}
                        <span className=' text-indigo-600 font-bold'>
                            Pacientes y Citas
                        </span>
                    </p>
                    {
                        pacientes.map(paciente => (
                            <Paciente
                                key={paciente.id}
                                paciente = {paciente}
                                setPacienteToEdit={setPacienteToEdit}
                                eliminarPaciente = {eliminarPaciente}
                            />
                        ))
                    }
                </>
            ) : (
                <>
                    <h2 className='font-black text-3xl text-center'>No hay pacientes</h2>
                    <p className='text-xl mt-5 mb-10 text-center'>
                        Cominenza agregando pacientes{' '}
                        <span className=' text-indigo-600 font-bold'>
                            y apareceran en este lugar
                        </span>
                    </p>
                </>
            ) }
        </div>
  )
}

export default ListadoPacientes
