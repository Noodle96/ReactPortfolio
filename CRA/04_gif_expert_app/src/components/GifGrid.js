import React, { Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({category}) => {
    const {data:imagesName,loading} = useFetchGifs(category);
    return (
        <Fragment>
            <h3 className='card animate__animated animate__backInDown'>{category}</h3>
            {loading && <p>Loading...</p>}
            <div className='card-grid'>
                
                {
                    imagesName.map(img => (
                        // <li key={img.id}>{img.title}</li>
                        <GifGridItem 
                            key={img.id}
                            //img={img}
                            {...img}
                            // title={img.title}
                            // url={img.url}
                        />
                    ))
                }
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}