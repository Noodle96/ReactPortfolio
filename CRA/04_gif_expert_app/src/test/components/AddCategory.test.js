

import React from 'react'; 
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { AddCategory } from '../../components/AddCategory';


describe('Pruebas en el componente <AddCategory />', () => {
    // const setCategory = () => {}
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory  setCategories={setCategories}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory  setCategories={setCategories}/>);
    });


    test('EL componente <AddCategory /> debería de mostrarse correctamente ', () => { 
        expect(wrapper).toMatchSnapshot();
    });

    test('Deberia de cambiar la caja de texto',() => {
        const input = wrapper.find('input');
        // console.log(input.html());
        const value = 'Hola mundo';
        input.simulate('change',{target:{value}});
        expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test('No debe de llamarse a la funcíon setCategories ', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });
    test('Deberia de llamarse el setcategories y limpiar la caja de text', () => {
        const input = wrapper.find('input');
        const value = 'pii';

        //SIMULAR EL INPUTCHANGE
        input.simulate('change',{target:{value}});
        expect(wrapper.find('p').text().trim()).toBe(value);

        //SIMULAR EL SUBMIT
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        // expect(setCategories).toHaveBeenCalled();
        // expect(setCategories).toHaveBeenCalledTimes(1); //the same
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        // console.log(wrapper.find('p').text().trim());
        //verificando que la caja de texto sea ''
        expect(wrapper.find('p').text().trim()).toBe('');
    });
});