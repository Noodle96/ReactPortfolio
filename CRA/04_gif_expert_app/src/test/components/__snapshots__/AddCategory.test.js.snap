// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> EL componente <AddCategory /> deber√≠a de mostrarse correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
